src/main/java/com/github/tomakehurst/wiremock/admin/model/RequestPatternTransformer.java

```diff
@@ -0,0 +1,72 @@
+package com.github.tomakehurst.wiremock.admin.model;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.github.tomakehurst.wiremock.http.ContentTypeHeader;
+import com.github.tomakehurst.wiremock.http.Request;
+import com.github.tomakehurst.wiremock.matching.MultiValuePattern;
+import com.github.tomakehurst.wiremock.matching.RequestPatternBuilder;
+import com.github.tomakehurst.wiremock.matching.StringValuePattern;
+import com.google.common.base.Function;
+
+import java.util.Map;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+
+/**
+ * Creates a RequestPatternBuilder from a Request's URL and method, and optionally headers from a whitelist.
+ * If headers patterns are supplied, the header will be only included in the RequestPatternBuilder if the predicate
+ * matches the request
+ */
+public class RequestPatternTransformer implements Function<Request, RequestPatternBuilder> {
+    private final Map<String, MultiValuePattern> headers;
+
+    @JsonCreator
+    public RequestPatternTransformer(@JsonProperty("headers") Map<String, MultiValuePattern> headers) {
+        this.headers = headers;
+    }
+
+    /**
+     * Returns a RequestPatternBuilder matching the URL and method of the Request. If header patterns are supplied,
+     * this will match them against the request and include them in the RequestPatternBuilder if there's a match.
+     */
+    @Override
+    public RequestPatternBuilder apply(Request request) {
+        final RequestPatternBuilder builder = new RequestPatternBuilder(request.getMethod(), urlEqualTo(request.getUrl()));
+
+        if (headers != null && !headers.isEmpty()) {
+            for (Map.Entry<String, MultiValuePattern> header : headers.entrySet()) {
+                String headerName = header.getKey();
+                MultiValuePattern matcher = header.getValue();
+                if (matcher.match(request.header(headerName)).isExactMatch()) {
```

Comment:
This behaviour is a bit of a departure from what we do currently e.g. use the value from the incoming request as the thing to match on, rather than a pre-determined value. I think this is probably more useful e.g. if you're recording an API the deals in more than one data content type then the recorder will understand that it needs to record different stubs for each `Accept` header value.

In the current recorder the operator is hard-coded to `equalTo`. What I had in mind here was being able to choose the operator, but still keep the right-hand operand from the request.